packages/react/src/internal/components/InputLabel.tsx

```diff
@@ -55,7 +59,7 @@ const InputLabel: React.FC<React.PropsWithChildren<Props>> = ({
       {required ? (
         <Box display="flex" as="span">
           <Box mr={1}>{children}</Box>
-          <span>*</span>
+          {requiredText && <span aria-hidden={!requiredIndicator ? true : undefined}>{requiredText}</span>}
```

Comment:
Right now `requiredIndicator` only hides the indicator to the accessibility tree (i.e. `aria-hidden="true"`). This is because the indicator is usually important in discerning which form fields are required visually, whereas it being present in the accessibility tree is not.

You can still technically hide the indicator by using an empty string in `requiredText` (e.g. `requiredText=""`).

Reply:
Do we want to allow an empty requiredText when the element is required 🤔 I _think_ I would be in favour here of that not being possible. I don't feel _super_ strongly about this but it could lead to audit issues in future if someone hides the requiredText but sets a field to be programatically required.

Reply:
That's a good point, I'm not entirely sure if there are instances today where we'd want to opt out of a visual indicator for required fields. The only instance that I could think of, is if someone were to opt for marking optional fields visually rather than required fields (e.g. `(optional)`). I don't think we have a case like that in GH today, so I'm okay with defaulting to '*' if `requiredText` is an empty string.